### PR TITLE
Cleaned up the nvFuser Python Frontend Batch Norm printing

### PR DESCRIPTION
**Pull request by kevinstephano**
* Removed `define_null_tensor` usage in favor of using optional arguments for binding.
* Re-ordered the non-State arguments for easier printing.
* Added a printing function to include booleans `training` and `channels_last`
* Fixed `define_tensor` to print `is_cpu`

cc @jjsjann123

### CODE REVIEW

File: torch/_prims/nvfuser_prims.py [hunk 1]
```diff
@@ -214,6 +214,8 @@ def _{fname}_nvfuser(fd, a, b, c):
 def _native_batch_norm_nvfuser(
     fd, input, weight, bias, running_mean, running_var, training, momentum, eps
 ):
+
+    """
```
Comment (IvanYashchuk) [inline torch/_prims/nvfuser_prims.py:218]: Let's remove this code

**Review by IvanYashchuk — APPROVED** [review]
Nice use of `c10::optional`!

File: torch/csrc/jit/codegen/cuda/python_frontend/python_bindings.cpp [hunk 2]
```diff
@@ -1292,26 +1283,38 @@ void initNvFuserPythonBindings(PyObject* module) {
...
-         nvfuser::Tensor running_var,
-         bool training,
+         nvfuser::Tensor arg,
+         c10::optional<nvfuser::Tensor> weight,
+         c10::optional<nvfuser::Tensor> bias,
+         c10::optional<nvfuser::Tensor> running_mean,
+         c10::optional<nvfuser::Tensor> running_var,
          nvfuser::Scalar momentum,
          nvfuser::Scalar eps,
+         bool training,
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/python_frontend/python_bindings.cpp:1293]: I have a mixed feeling here. Why do we not want to keep the same arg order as with `aten::batch_norm`?

**Review by jjsjann123 — APPROVED** [review]
minor comments.
LGTM

File: torch/csrc/jit/codegen/cuda/python_frontend/python_bindings.cpp (hunk 2 above)
Comment (kevinstephano) [reply torch/csrc/jit/codegen/cuda/python_frontend/python_bindings.cpp:1293]: The reason I had to re-order the params is because `training` and `channels_last` were not defined as `State`.  `State` objects are mechanically added in front of any Fusion Record specific attributes.  I chose to do the simplest, thing.  That was to re-order the parameters.  All the choices:

1. Reorder parameters `State` in front of the Fusion Record Attributes.
2. Make an explicit list of the list of items to be printed to allow for intermixing.
3. Turn `training` and `channels_last` into state.

Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/python_frontend/python_bindings.cpp:1293]: > `State` objects are mechanically added in front of any Fusion Record specific attributes.

But that's in fusion record right?

For exposed python binding, I thought the order of arguments doesn't affect how we construct IRs in fusion record and this should be just a mechanical change.

Comment (kevinstephano) [reply torch/csrc/jit/codegen/cuda/python_frontend/python_bindings.cpp:1293]: It impacts the printing. The printing mechanically goes through the `args` first.  So, if you try to re-use the printed function and `training` is in the middle of the `State` args, it will fail.

It was a choice not to make `training` and `channels_last` state args.

https://github.com/pytorch/pytorch/blob/139afc50ecafcde5fb085e2cca78fed55e6b5aad/torch/csrc/jit/codegen/cuda/python_frontend/fusion_record.h#L1451

https://github.com/pytorch/pytorch/blob/139afc50ecafcde5fb085e2cca78fed55e6b5aad/torch/csrc/jit/codegen/cuda/python_frontend/fusion_record.h#L140-L154

Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/python_frontend/python_bindings.cpp:1293]: Sorry if I'm trying too hard with nitpicking :stuck_out_tongue: 

That requirement seems to only apply to `RecordFunctor` though.

I was suggesting that our python API to match aten signature. `nvf_ops.def("batch_norm", ..., bool training, nvfuser::Scalar momentum, nvfuser::Scalar eps)`
But later in the record definition, we can still do
```
        fd->defineRecord(new nvfuser::BatchNormOpRecord(
            {fd->recordingState(arg()),
             weight_state,
             bias_state,
             running_mean_state,
             running_var_state,
             fd->recordingState(momentum()),
             fd->recordingState(eps())},
            {fd->recordingState(output()),
             fd->recordingState(mean()),
             fd->recordingState(invstd())},
            training,
            channels_last));
```

**Review by mruberry — APPROVED** [review]
Stamped! So long as @IvanYashchuk and @jjsjann123 are happy
